python/cuml/test/test_umap.py

```diff
@@ -12,50 +12,84 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 #
-import pytest
 
-from cuml.manifold.umap import UMAP
 
+# Please install UMAP before running the code
+# use 'conda install -c conda-forge umap-learn' command to install it
+
+import pytest
+from cuml.test.utils import array_equal
+
+from cuml.manifold.umap import UMAP as UMAP_cuml
+import umap
 import cudf
 import pandas as pd
 import numpy as np
-
 from sklearn import datasets
```

Comment:
Does this just crash if umap-learn is not installed? Is that a mandatory package?
One neat trick is that _inside each test_ you can do: ```
   umap = pytest.importorskip("umap")
```
and it will mark the test to be skipped if the `umap` package is not installed. https://docs.pytest.org/en/latest/skipping.html 
```suggestion
from sklearn import datasets
```
```suggestion
from sklearn import datasets
```

Reply:
yes, it is a mandatory package since we are comparing the results of cuML umap with the original umap